### PR TITLE
Gateway 3.5 Autodocs

### PR DESCRIPTION
**Pull request by Guaris**
Contains CLI, PDK, and OSS+EE API spec changes.

### CODE REVIEW

File: app/_src/gateway/reference/cli.md [hunk 1]
```diff
@@ -271,6 +374,10 @@ Options:
  --db-timeout     (default 60)
  --lock-timeout   (default 60)
 
+{% if_version gte:3.3.x %}
```
Comment (Guaris) [inline app/_src/gateway/reference/cli.md:377]:
```suggestion
{% if_version gte:3.5.x %}
```

File: app/_src/gateway/reference/cli.md [hunk 2]
```diff
@@ -246,6 +347,8 @@ Options:
  -c,--conf        (optional string) configuration file
  -p,--prefix      (optional string) prefix Kong is running at
  --nginx-conf     (optional string) custom Nginx configuration template
+ --nginx-conf-flags        (optional string) flags that can be used to control
+                                             how Nginx configuration templates are rendered
```
Comment (lena-larionova) [inline app/_src/gateway/reference/cli.md:351]:
```suggestion
{% if_version gte:3.5.x %}
 --nginx-conf-flags        (optional string) flags that can be used to control
                                             how Nginx configuration templates are rendered
{% endif_version %}
```

File: api-specs/Gateway-EE/3.5/kong-ee-3.5.yaml [hunk 3]
```diff
@@ -328,6 +328,34 @@ components:
           tags:
             - silver-tier
           username: bob-the-builder
+    HMAC-auth-credential:
```
Comment (lena-larionova) [inline api-specs/Gateway-EE/3.5/kong-ee-3.5.yaml:331]: These are enabled by the HMAC auth plugin - are you okay with adding them into these specs?

Comment (Guaris) [reply api-specs/Gateway-EE/3.5/kong-ee-3.5.yaml:331]: Do you think this opens us up to requests for other plugin endpoints, if so I'll remove it and open a post 3.5 PR with more of the plugin endpoints that are unrelated to 3.5. I have a good way to add those in now.

Comment (lena-larionova) [reply api-specs/Gateway-EE/3.5/kong-ee-3.5.yaml:331]: If you want to add the rest of the plugins, then go for it. I mostly wanted to check if this was appearing in here on purpose, and if it was, then we open the door for adding all the other plugin endpoints.

**Review by lena-larionova — APPROVED** [review]

**Pull request merged**